clients/src/test/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImplTest.java

```diff
@@ -728,7 +722,7 @@ public void testDelayedReconciliationResultAppliedWhenTargetChangedWithMetadataU
 
         // Member should update the subscription and send ack when the delayed reconciliation
         // completes.
-        verify(subscriptionState).assignFromSubscribed(Collections.emptySet());
+        verify(subscriptionState).assignFromSubscribedAwaitingCallback(Collections.emptySet(), Collections.emptySet());
```

Comment:
Why do we verify something else here? This PR only updates test code, so where does this change in behavior come from? Seems I am missing something.